### ISSUE TITLE
Test for side effects in Metafeatures.compute function

### ISSUE THREAD

**Issue by bjschoenfeld** [byu-dml/metalearn#34]
There can be (at least) two kinds of side effects from the compute function:

1. The input data is modified after it is passed to compute. This can happen if the code writes to the mutable input data.
2. The functionality of compute changes between subsequent calls of the same instance of Metafeatures due to some persisting internal state.

I suspect that these are not happening, but they are good test cases.

**Comment by macetheace96** [comment]
how many times should we run compute in order to test part 2? Is just twice enough?

**Comment by bjschoenfeld** [comment]
Twice should be enough. If there is any persisting state, it should present itself after the first run.

**Comment by macetheace96** [comment]
This could potentially be good to add as a more general check for all tests similar to #83 so we can see if any particular parameter has a persisting state.

**Comment by bjschoenfeld** [comment]
Yes, factoring it out into its own function would make it easy to use later.
